pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -118,7 +118,7 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	}
 
 	nodeFilter := []ast.Node{
-		(*ast.TypeSpec)(nil),
+		(*ast.GenDecl)(nil),
```

Comment:
nit: 
I stopped to think about what the difference between a generic declaration / type declaration was and why this was changed as it wasn't immediately obvious to me.

From the docs: 
(A GenDecl node (generic declaration node) represents an import, constant, type or variable declaration. A valid Lparen position (Lparen.IsValid()) indicates a parenthesized declaration. ) 

I'd include the *why* in the PR description here?, I think it's just to allow us to extract comments on the additional nodes (e.g variable or constant decl),  rather than just type decls.

Reply:
Yeah fair, Bryce and I had discussed this one before so we both have context

If you declare a type as below, then the comments are on the GenDecl, type is the declaration, and it has within it a TypeSpec (the word `struct` onwards here)
```
// Comment line
type Foo struct {}
```
If however you do it as in this second example, the comments would be on the TypeSpec itself.
```
type {
  // Comment line
  Foo struct{}
}
```
Bryce and I concluded that usage of the second declaration type is uncommon, and for now we would not considered it supported, until we see others using it, at which point we can make a call on carrying the complexity of supporting both

Reply:
Might be worth taking the comment from https://github.com/JoelSpeed/kal/pull/2/files#diff-4bd55f5eca464c04c911c344bac6f102e033a5bd1b6578ce5503af8749209778R121-R137 and adding it here to be more explicit for someone looking at the code in the future?

Reply:
Yep, that was a really well written comment, have copied across